src/Application/FacetConfigList.php

```diff
@@ -20,14 +20,14 @@ public function __construct( FacetConfig ...$facets ) {
 	}
 
 	private function addFacetConfig( FacetConfig $facetConfig ): void {
-		$this->facetsPerItemId[$facetConfig->itemId->getSerialization()][] = $facetConfig;
+		$this->facetsPerInstanceType[$facetConfig->instanceTypeId->getSerialization()][] = $facetConfig;
```

Comment:
There is arguably some technical debt here, but it's fully contained, so it does not matter.

Two guidelines violated here:
* Compute things when you need them, don't put them in place just in case you need them later
* Do not do work in the constructor

Reply:
The original idea was that we'd need to access per-item/instance facet config when building the actual facets. So to build FacetConfigList I had to decide how to keep track of the configs and that was the first use case. However, in https://github.com/ProfessionalWiki/WikibaseFacetedSearch/issues/31 I'm going to need the facet configs but without caring which items/instances they are for - which results in doing `array_values()` on this item-indexed array. So it ends up requiring additional processing either way.

Probably we should just put them in a plain list and have something like `getFacetsByInstanceType()` that returns them indexed by item/instance ID.